### PR TITLE
macOS add signing of DMG and ZIPed export

### PR DESCRIPTION
**Pull request by bruvzg**
1. Sign `DMG` after exporting (required for notarizing).
2. Sign `.app` bundle when exporting as `ZIP` on macOS.

Instead of separate export code that was modifying copy of export template `ZIP`, it now reuses same export procedure for both `ZIP` and `DMG` - extracting export template, signing and creating new archive.

### CODE REVIEW

File: platform/osx/export/export.cpp [hunk 1]
```diff
@@ -670,122 +634,129 @@ Error EditorExportPlatformOSX::export_project(const Ref<EditorExportPreset> &p_p
...
+					Z_DEFLATED,
+					Z_DEFAULT_COMPRESSION,
+					0,
+					-MAX_WBITS,
+					DEF_MEM_LEVEL,
+					Z_DEFAULT_STRATEGY,
+					NULL,
+					0,
+					0x0314, // "version made by", 0x03 - Unix, 0x14 - ZIP specification version 2.0, required to store Unix file permissions
+					0);
```
Comment (bruvzg) [inline platform/osx/export/export.cpp]: Note: Same code for setting `ZIP` version info (required for setting executable permissions), might be also useful for #527, AFAIK it's still not resolved.

**Comment by akien-mga** [comment]
Should be good to merge once rebased.

**Comment by aaronfranke** [comment]
This needs another rebase.

**Comment by bryantfhayes** [comment]
Is this going to target 4.x release?

**Comment by Calinou** [comment]
@bryantfhayes Probably, but pull request reviews may take a while as the `master` branch is still being refactored.

File: platform/osx/export/export.cpp (hunk 1 above)
Comment (akien-mga) [reply platform/osx/export/export.cpp:770]: If you want to give it a try, that'd be awesome.

I tried to solve this a while ago but I couldn't get permissions to be kept in the Zip whatever I tried. Zip documentation is also abysmal when it comes to Unix permissions.

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.2.2.

**Pull request merged**